### PR TITLE
[Term Entry] Dart List .join()

### PR DESCRIPTION
**Pull request by GregDev800**
### Description

- Added the .join() method term for Dart
- Closes issue #4311

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/dart/concepts/list/terms/join/join.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'A tag'
+  - 'A second tag'
+  - 'An nth tag'
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-example-course'
+  - 'paths/example-path'
+---
+
+# **join**
```
Comment (Vino-Siva) [inline content/dart/concepts/list/terms/join/join.md:17]: Hi, can you change this to `h2` header? Please read the documentation [Headers Style Guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md#headerssections) for further details.

Comment (GregDev800) [reply content/dart/concepts/list/terms/join/join.md]: Hello, done.

File: content/dart/concepts/list/terms/join/join.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  - 'A second subject name'
+  - 'An nth subject name'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'A tag'
+  - 'A second tag'
+  - 'An nth tag'
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-example-course'
+  - 'paths/example-path'
+---
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:15]: Please fill in the details here. You can find the listed [subjects](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md), [tags](https://github.com/Codecademy/docs/blob/main/documentation/tags.md), and [catalog content](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md). Check out the [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) if you want.

File: content/dart/concepts/list/terms/join/join.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+## **join**
+&nbsp;  
+&nbsp;  
+
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:21]: The entry doesn't need this to be included.

File: content/dart/concepts/list/terms/join/join.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+&nbsp;  
+
+## Syntax
+
+```
+list-name.join(separator); // the raw syntax
+
+chores.join(" + "); // a real example
+```
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:32]: The syntax should be enclosed inside ` ```pseudo` block.

File: content/dart/concepts/list/terms/join/join.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+The method refers to its object, in the below case, to the 'chores' variable, which holds list (or array) elements. Inside the brackets '( )' there's a parameter - a separator containing any sign/-s wrapped with single/double quotes. The main role of a separator is, as the name suggests, separating elements from one another - allowing for better readability and additional or new meaning to the text. Spaces added to a separator are nonobligatory.
+
+&nbsp;
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:36]:
```suggestion

```

File: content/dart/concepts/list/terms/join/join.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+The method refers to its object, in the below case, to the 'chores' variable, which holds list (or array) elements. Inside the brackets '( )' there's a parameter - a separator containing any sign/-s wrapped with single/double quotes. The main role of a separator is, as the name suggests, separating elements from one another - allowing for better readability and additional or new meaning to the text. Spaces added to a separator are nonobligatory.
+
+&nbsp;  
+
+
+## Example No 1
+
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:41]: Remove Extra space from the entry.

File: content/dart/concepts/list/terms/join/join.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+print(planOfDut); // returns:  planting flowers (sigh) vacuuming (sigh) laundry (sigh) ironing (sigh) preparing meals
+
+
+```
+&nbsp;  
+
+
+## Example No 2
+
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:83]: Remove extra space.

File: content/dart/concepts/list/terms/join/join.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+### Tags
+
+```
+Concatenation
+Dart
+Lists
+
+```
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:108]: Tags should not be included here. Please follow this [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md).

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hello, as your entry lacks many things I've suggested the changes. Please make them as I instructed. I see that this is your first time contributing here but don't worry check out our [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md) and then make the changes. All the best!!

**Comment by GregDev800** [comment]
Hi, some changes have been made.

File: content/dart/concepts/list/terms/join/join.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
+---
+Title: 'join'
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:2]:
```suggestion
Title: '.join()' 

```

File: content/dart/concepts/list/terms/join/join.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
+---
+Title: 'join' 
+Description: 'The join() method concatenates (combines) list elements as a string including various separators between them dependent on a user's need.'
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:3]:
```suggestion
Description: 'Concatenates list elements as a string.'
```

File: content/dart/concepts/list/terms/join/join.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+  - 'Web Development'
+Tags:
+  - 'Concatenation'
+  - 'Dart'
+  - 'Lists'
+CatalogContent:
+  - 'paths/computer-science'
+---
+
+The **`join()`** method concatenates (*combines*) list elements as a string including various separators between them dependent on a user's need.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:15]:
```suggestion
The **`.join()`** method concatenates (combines) list elements as a string including various separators between them dependent on a user's need.

```

File: content/dart/concepts/list/terms/join/join.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+## Syntax
+
+```pseudo
+list-name.join(separator); // the raw syntax
+
+chores.join(" + "); // a real example
+```
+
+The method refers to its object, in the below case, to the 'chores' variable, which holds list (or array) elements. Inside the brackets '( )' there's a parameter - a separator containing any sign/-s wrapped with single/double quotes. The main role of a separator is to separate elements from one another - allowing for better readability and additional or new meaning to the text. Spaces added to a separator are nonobligatory.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:26]: As I've removed the example from the syntax, remove the description of it too. You can describe the parameters using the following format - 
```
- `parameter`:  Describe the parameter.
```

File: content/dart/concepts/list/terms/join/join.md [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:27]:
```suggestion

```

File: content/dart/concepts/list/terms/join/join.md [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+## Example 1
+
+The following block codes are showing an instance of removed spaces between elements, however a default `,` remains:
+
+```dart
+List<String> tought_to_read = ["A", "bit", "hard", "to", "read", "innit", "?"];
+
+String try_to_read = tought_to_read.join(); // Spaces between elements are removed.
+print(try_to_read);
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:37]: The whole code block should be executable, you need to wrap the code inside the main function (Do this for every code block).

File: content/dart/concepts/list/terms/join/join.md [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+```dart
+List<String> tought_to_read = ["A", "bit", "hard", "to", "read", "innit", "?"];
+
+String try_to_read = tought_to_read.join(); // Spaces between elements are removed.
+print(try_to_read); 
+```
+The output of `.join()`:
+
+```shell
+A,bit,hard,to,read,innit,?
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:42]: You have not specified the separator, the output will not be the same. Please check it.

File: content/dart/concepts/list/terms/join/join.md [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+
+The following block codes are showing an instance of removed spaces between elements, however a default `,` remains:
+
+```dart
+List<String> tought_to_read = ["A", "bit", "hard", "to", "read", "innit", "?"];
+
+String try_to_read = tought_to_read.join(); // Spaces between elements are removed.
+print(try_to_read); 
+```
+The output of `.join()`:
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:39]:
```suggestion

The output of the above code is:

```
A line like this should be instead of the current one and remember to give the one-line space before and after a code block (Do this throughout the entire entry).

File: content/dart/concepts/list/terms/join/join.md [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+```
+The output of `.join("")`:
+
+```shell
+You are awesome!
+```
+
+## Example 3
+
+The below code blocks are presenting an example of using special signs, that can be altered:
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:63]:
```suggestion
In the following example, the asterisk is used as a delimiter between elements.

```

File: content/dart/concepts/list/terms/join/join.md [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+```
+The output of `.join(" * ")`:
+
+```shell
+JavaScript * Python * HTML * CSS
+```
+
+## Example 4
+
+The following fragments are showing an example of a conjunction, which can be also replaced with any:
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:79]:
```suggestion
In the following example, the separator used between each element is `or` meaning each task in the list will be separated by the word `or`.

```

File: content/dart/concepts/list/terms/join/join.md [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+```
+The output of `.join(" or ")`:
+
+```shell
+planting flowers or vacuuming or laundry or ironing or preparing meals
+```
+
+## Example 5
+
+The below snippet is showing another list but combined with a predefined text to make it slightly more complex, plus the output:
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:95]:
```suggestion
The following snippet contains a list which has a predefined text to make it slightly more complex:

```

**Review by avdhoottt — APPROVED** [review]
@GregDev800 Please make the suggested changes.

File: content/dart/concepts/list/terms/join/join.md (hunk 9 above)
Comment (GregDev800) [reply content/dart/concepts/list/terms/join/join.md:2]: How about this fragment, it says differently?
metadata

File: content/dart/concepts/list/terms/join/join.md (hunk 14 above)
Comment (GregDev800) [reply content/dart/concepts/list/terms/join/join.md:37]: All lists and variables are global ones and they don't need another function, except print() here, to be executed.

File: content/dart/concepts/list/terms/join/join.md (hunk 12 above)
Comment (GregDev800) [reply content/dart/concepts/list/terms/join/join.md:26]: I completely do not understand your request. Could you rephrase, please? Rather than suggesting lots of changes, is it possible to write down the reasons behind them?

Comment (GregDev800) [reply content/dart/concepts/list/terms/join/join.md:26]: I thought that there must be also a description of syntax in general, not a specific parameter, because it would not be logical.

**Comment by GregDev800** [comment]
Changes incorporated.

File: content/dart/concepts/list/terms/join/join.md [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+  - 'Concatenation'
+  - 'Dart'
+  - 'Lists'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+The **`.join()`** method concatenates (combines) list elements as a string including various separators between them dependent on a user's need.
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:17]:
```suggestion
 
```

File: content/dart/concepts/list/terms/join/join.md [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+The **`.join()`** method concatenates (combines) list elements as a string including various separators between them dependent on a user's need.
+ 
+
+## Syntax
+
+```pseudo
+list-name.join('separator')
+```
+ - `separator`: The parameter which separates elements in a list by using any characters wrapped with single/double quotes. White spaces can be used but aren't obligatory.
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:24]:
```suggestion

 - `separator`: It separate elements in a list using any characters wrapped with single/double quotes. White spaces can be used but aren't obligatory.

```

File: content/dart/concepts/list/terms/join/join.md [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+
+## Example 1
+
+The next snippet depicts an occurrence of erased separators entirely:
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:28]:
```suggestion
The following code snippet demonstrates the `.join` method:

```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Some final changes.

File: content/dart/concepts/list/terms/join/join.md [hunk 23]
```diff
@@ -0,0 +1,106 @@
...
+```pseudo
+list-name.join('separator')
+```
+
+ - `separator`: It separate elements in a list using any characters wrapped with single/double quotes. White spaces can be used but aren't obligatory.
+
+## Example 1
+
+The following code snippet demonstrates the `.join` method:
+
```
Comment (avdhoottt) [inline content/dart/concepts/list/terms/join/join.md:29]:
```suggestion
```

**Review by avdhoottt — APPROVED** [review]
LGTM!!

File: content/dart/concepts/list/terms/join/join.md (hunk 22 above)
Comment (GregDev800) [reply content/dart/concepts/list/terms/join/join.md]: > It shows a separator with no white spaces.

File: content/dart/concepts/list/terms/join/join.md (hunk 23 above)
Comment (GregDev800) [reply content/dart/concepts/list/terms/join/join.md:29]: > A line like this should be instead of the current one and remember to give the one-line space before and after a code block (Do this throughout the entire entry).

I was supposed to leave a one-break line as you mentioned.

**Comment by GregDev800** [comment]
Changes applied.

File: content/dart/concepts/list/terms/join/join.md [hunk 24]
```diff
@@ -0,0 +1,104 @@
...
+The **`.join()`** method concatenates (combines) list elements as a string including various separators between them dependent on a user's need.
+
+## Syntax
+
+```pseudo
+list-name.join('separator')
+```
+ - `separator`: It separates elements in a list using any characters wrapped with single/double quotes. White spaces can be used but aren't obligatory.
+
+## Example 1
```
Comment (ishg-153) [inline content/dart/concepts/list/terms/join/join.md:25]:
```suggestion
## Example
```

File: content/dart/concepts/list/terms/join/join.md [hunk 25]
```diff
@@ -0,0 +1,104 @@
...
+} 
+```
+
+The output of the above code is:
+
+```shell
+You are awesome!
+```
+
+## Example 2
```
Comment (ishg-153) [inline content/dart/concepts/list/terms/join/join.md:45]:
```suggestion
```

File: content/dart/concepts/list/terms/join/join.md [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+} 
+```
+
+The output of the above code is:
+
+```shell
+JavaScript * Python * HTML * CSS
+```
+
+## Example 3
```
Comment (ishg-153) [inline content/dart/concepts/list/terms/join/join.md:65]:
```suggestion
```

File: content/dart/concepts/list/terms/join/join.md [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+}
+```
+
+The output of the above code is:
+
+```shell
+planting flowers or vacuuming or laundry or ironing or preparing meals
+```
+
+## Example 4
```
Comment (ishg-153) [inline content/dart/concepts/list/terms/join/join.md:85]:
```suggestion
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@GregDev800 I have suggested removing the separate example subheadings and just keeping a common one. Please make the changes and let me know once done.

**Comment by GregDev800** [comment]
> @GregDev800 I have suggested removing the separate example subheadings and just keeping a common one. Please make the changes and let me know once done.

Done.

**Review by ishg-153 — APPROVED** [review]
@GregDev800 Great! This looks ready to merge.
